plugins/board-resources/src/components/MoveCard.svelte

```diff
@@ -0,0 +1,124 @@
+  import { getClient } from '@anticrm/presentation'
+  import { Class, Client, Doc, Ref } from '@anticrm/core'
+  import { getResource, OK, Resource, Status } from '@anticrm/platform'
+  import { Card } from '@anticrm/board';
+  import view from '@anticrm/view'
+  import board from '../plugin'
+  import SpaceSelect from './move-selectors/SpaceSelect.svelte'
+  import StateSelect from './move-selectors/StateSelect.svelte'
+  import RankSelect from './move-selectors/RankSelect.svelte'
+
+  export let object: Card
+
+  const client = getClient()
+  const hierarchy = client.getHierarchy()
+  const dispatch = createEventDispatcher()
+  let status: Status = OK
+  const selected = {
+    space: object.space,
+    state: object.state,
+    rank: object.rank
+  }
+
+  async function move (doc: Card): Promise<void> {
+    client.updateDoc(doc._class, doc.space, doc._id, selected)
+    dispatch('close')
+  }
+
+  async function invokeValidate (
+    action: Resource<<T extends Doc>(doc: T, client: Client) => Promise<Status>>
+  ): Promise<Status> {
+    const impl = await getResource(action)
+    return await impl(object, client)
+  }
+
+  async function validate (doc: Doc, _class: Ref<Class<Doc>>): Promise<void> {
+    const clazz = hierarchy.getClass(_class)
+    const validatorMixin = hierarchy.as(clazz, view.mixin.ObjectValidator)
+    if (validatorMixin?.validator != null) {
+      status = await invokeValidate(validatorMixin.validator)
+    } else if (clazz.extends != null) {
+      await validate(doc, clazz.extends)
+    } else {
+      status = OK
+    }
+  }
+
+  $: validate({...object, ...selected}, object._class)
+</script>
+
+<div class="container">
+  <div class="overflow-label fs-title">
+    <Label label={board.string.MoveCard}/>
+  </div>
+  <StatusControl {status} />
+  <div class="content-accent-color mt-4 mb-4">
+    <Label label={board.string.SelectDestination} />
+  </div>
+  <div class="spaceSelect">
+    <SpaceSelect label={board.string.Board} object={object} bind:selected={selected.space} />
+  </div>
+  <div class="spaceSelect">
+    {#key selected.space}
+      <StateSelect label={board.string.List} object={object} space={selected.space} bind:selected={selected.state} />
+    {/key}
+  </div>
+  <div class="spaceSelect">
+    {#key selected.state}
+      <RankSelect label={board.string.Position} object={object} state={selected.state} bind:selected={selected.rank} />
+    {/key}
+  </div>
+  <div class="footer">
+    <Button
+      label={board.string.Move}
+      size={'small'}
+      disabled={status !== OK || object.state === selected.state && object.rank === selected.rank}
+      kind={'primary'}
+      on:click={() => {
+        move(object)
+      }}
+    />
+    <Button
+      size={'small'}
+      label={board.string.Cancel}
+      on:click={() => {
+        dispatch('close')
+      }}
+    />
+  </div>
+</div>
+
+<style lang="scss">
+  .container {
+    display: flex;
```

Comment:
Can we reuse global styles from `themes/packages/styles/_layouts.scss` as much as possible here?

Reply:
Done